tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs

```diff
@@ -79,30 +78,39 @@ public static void Constructor_VerifyGuardClauses()
         [Test]
         [InMemoryDbAutoData]
         public static async Task UpdateAsync_OrderDoesNotExist_ReturnsNotFound(
+            [Frozen] Mock<IOrderingPartyService> service,
             CallOffId callOffId,
             OrderingPartyModel model,
             OrderingPartyController controller)
         {
+            service.Setup(o => o.GetOrder(callOffId)).ReturnsAsync((Order)null);
+
             var response = await controller.UpdateAsync(callOffId, model);
+            service.Verify(o => o.GetOrder(callOffId));
 
             response.Should().BeOfType<NotFoundResult>();
         }
 
         [Test]
         [InMemoryDbAutoData]
         public static async Task UpdateAsync_UpdatesOrderingParty(
-            [Frozen] ApplicationDbContext context,
+            [Frozen] Mock<IOrderingPartyService> service,
             [Frozen] CallOffId callOffId,
             Order order,
             OrderingPartyModel model,
             OrderingPartyController controller)
         {
             order.OrderingParty.Should().NotBeEquivalentTo(model);
-
-            context.Order.Add(order);
-            await context.SaveChangesAsync();
+            service.Setup(o => o.GetOrder(callOffId)).ReturnsAsync(order);
+            service.Setup(o => o.SetOrderingParty(order, It.IsAny<OrderingParty>(), It.IsAny<Contact>())).Callback(() =>
+            {
+                order.OrderingParty.Name = model.Name;
+                order.OrderingParty.OdsCode = model.OdsCode;
+            });
 
             await controller.UpdateAsync(callOffId, model);
+            service.Verify(o => o.GetOrder(callOffId));
+            service.Verify(o => o.SetOrderingParty(order, It.IsAny<OrderingParty>(), It.IsAny<Contact>()));
```

Comment:
As above, I don't think we need to `Verify` tests here as the test covers this as part of its assertion.

Reply:
Removed `service.Verify()`